scripts/test-gem-build

```diff
@@ -23,6 +23,7 @@ if [[ "${BUILD_NATIVE_GEM}" == "ruby" ]] ; then
   bundle exec rake clean
   bundle exec rake gem
 else
+  export MACOSX_DEPLOYMENT_TARGET=10.14
```

Comment:
I tried hard to avoid this duplication in the `Rakefile`. I originally tried to pass `cmake` `-DCMAKE_CXX_FLAGS=-mmacosx-version-min=10.14` (https://github.com/tpoechtrager/osxcross#deployment-target), but the compiler flags had redundant values, with the latter one overriding the one I set:

```
-mmacosx-version-min=10.14 -mmacosx-version-min=10.13 ...
```

Reply:
Does this mean the native gems for macOS will no longer run on High Sierra (10.13) and instead require Mojave (10.14)? If so, perhaps we should add that to the README along with the glibc requirements.

Reply:
Yes. The alternative is to file an upstream fix to abseil to be able control the behavior of `ABSL_OPTION_USE_STD_OPTIONAL` independently of `MACOSX_DEPLOYMENT_TARGET`, but I suspect it's not worth the effort.